[Nguyen Doan Phuong Anh] iP

april-anh opened the pull request:
# CS2103T Chatbot 
### (**previously named _Duke_**)

This chatbot helps you to:
- Manage your to-do list
- Use the command line so that you can get rid of the mouse

Basic features:
1. Add tasks
2. Delete tasks
3. Mark / Unmark as done
4. Add deadlines

If you are learning Java, let's get started with the `main` method:
```Java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

All you have to do are:
- [x] Install Java
- [ ] Download chatbot from this repo
- [ ] Use it :+1: :100:

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Commit:
Add Gradle support

Commit:
Adding constant and method in level 1

Commit:
Add documentation for level 1

Commit:
add list to store text typed by the user, "add" method and "list" method

Commit:
mark as done

Commit:
todo, deadline. event

Commit:
edit test case for text-ui-test

Commit:
handle input exception

Commit:
change data structure from array to array list and add delete method

Commit:
add more input exception handle for (un)mark and delete

Commit:
add method rewriteList

Commit:
Level 7: save

Commit:
Merge branch 'branch-Level-7'

Commit:
Reformat deadline date

Commit:
Merge branch 'branch-Level-8'

Commit:
Moving add and delete task method to package

Commit:
move list method to package

Commit:
Move process method to parsre

Commit:
put all class in to duke package

Commit:
Merge remote-tracking branch 'upstream/add-gradle-support'

Commit:
add unit test

Commit:
update saved list

Commit:
update List.txt position

Commit:
add javadoc

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
delete unsed method

Commit:
change indentation of switch case

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
add find command

Commit:
Merge branch 'branch-Level-9'

edricyeo commented on src/main/java/duke/Storage/DukeDecoder.java:
```diff
@@ -0,0 +1,41 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class DukeDecoder {
```
Could include JavaDoc statement for the class.

edricyeo commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,39 @@
+import duke.Storage.DukeDecoder;
+import duke.Parser.ProcessUserInput;
+import Ui.Constants;
+import duke.TaskList.*;
```
Could list the import statements from the TaskList package explicitly.

edricyeo commented on src/main/java/duke/Exception/DukeException.java:
```diff
@@ -0,0 +1,83 @@
+     */
+    public static class InvalidInputException extends DukeException {
+        public InvalidInputException() {
+            super("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
```
I like how you nested various exception classes in the DukeException class! Perhaps you could include the "☹ OOPS!!!" statement in the DukeException class, so that you do not need to repeat it for every new exception.

edricyeo commented on src/main/java/duke/Parser/ProcessUserInput.java:
```diff
@@ -0,0 +1,139 @@
+                break;
+            case Constants.UNMARK:
+                try {
+                    userInput.substring(8);
```
Perhaps you could consider using the .split() method instead of the .substring() method? I think it may be more reusable since you would not need to manually change the value of the substring length based on the keyword.

edricyeo commented on src/main/java/duke/TaskList/Task.java:
```diff
@@ -0,0 +1,76 @@
+        return this.getStatusIcon() + " " + this.detail;
+    }
+
+    protected String statusIcon() {
```
Could include JavaDocs for this method.

edricyeo commented on src/main/java/duke/Ui/Constants.java:
```diff
@@ -0,0 +1,54 @@
+package Ui;
+
+public class Constants {
```
Could include JavaDocs for this class maybe.

edricyeo reviewed (commented):
Overall good effort! Just a few style and JavaDocs improvements that could be made! 👍

Commit:
Creating basic interface

Commit:
Adding custom behavior to DialogBox

Commit:
trial JavaFX tutorial 4 - Rebuilding the Scene using FXML

Commit:
Change method signature to fit GUI (for process userInput from void to String)

Commit:
Complete process user input method

Commit:
Merge branch 'Level-10'

Commit:
Configure checkstyle

Commit:
Fix coding standard according to checkstyle.

Commit:
Merge branch 'A-CheckStyle'

Commit:
Reorganize package, add greeting message to GUI.

debwy commented on src/main/java/duke/logic/task/Task.java:
```diff
@@ -0,0 +1,84 @@
+     */
+    public String storedData() {
+        return statusIcon() + "|" + detail;
+    }
```
Perhaps it's better to rename these methods as verbs?

debwy commented on src/test/java/duke/task/TodoTest.java:
```diff
@@ -0,0 +1,13 @@
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class TodoTest {
```
You might want to match the capitalisation of the ToDo class?

debwy commented on src/main/java/duke/logic/task/Task.java:
```diff
@@ -0,0 +1,84 @@
+        this.isDone = isDone;
+    }
+    /**
+     * Get icon for status done or not
```
Maybe you could rename javadoc method summaries in the form "Gets..." or "Marks..."?
You could also put an empty space between the description and parameter sections?

debwy commented on src/main/java/duke/exception/DukeException.java:
```diff
@@ -0,0 +1,93 @@
+ * Class for exception from user input.
+ */
+public class DukeException {
+    /**
+     * Error message for exception.
+     */
```
Maybe javadoc of class members could be specified on a single line? Not a java coding standard requirement but it might make the code easier to read

debwy commented:
As a whole I think your code is really clean! Really liked how you sectioned your classes (e.g. splitting both decoder & encoder classes) and how thorough the exceptions were (such that most of the comments were just nitpicks), nice job 👍

Commit:
Build a jar file for GUI version

Jar file created by IntelliJ.

Jar file doesn't have GUI.

Let's use Gradle shadow plug-in to create Jar file.

Gradle shadow includes JavaFX library.

Refer to https://se-education.org/guides/tutorials/gradle.html#creating-a-jar-file for guidance.

Commit:
Add assertion to check user's bad input

april-anh committed:
Merge pull request #1 from april-anh/A-Assertions

Add assertion to check user's bad input

Commit:
Merge branch 'master' of https://github.com/april-anh/ip

Commit:
Merge branch 'master' into Level-10

Commit:
Remove unnecessary comments

Commit:
Refactor ProcessUserInput class: move exception handler to class Task

Commit:
Refactor ProcessUserInput class: unmark method

Rename to markAsNotDone

move exception handler to class Task

Commit:
Refactor method add

Avoid long method

Commit:
Refactor delete method

Commit:
Delete TaskOperation class:

Commit:
Refactor method find

april-anh committed:
Merge pull request #2 from april-anh/heads/A-CodeQuality

Heads/a code quality

Commit:
Merge branch 'master' of https://github.com/april-anh/ip

Commit:
Fix checkstyle

Commit:
Merge branch 'branch-Level-10'

Commit:
Uncomment assert statement

april-anh committed:
Merge pull request #3 from april-anh/A-CodeQuality

Fix wrong naming for branch A-CodeQuality

Commit:
Merge branch 'master' of https://github.com/april-anh/ip into A-CodeQuality

Commit:
Merge branch 'master' of https://github.com/april-anh/ip

Commit:
Merge branch 'branch-A-CodeQuality'

Commit:
Improve find method in Task class:

Support find partial keywords

Commit:
Merge branch 'branch-C-BetterSearch'

Commit:
Add exit command

Commit:
Fix bug cannot save file in jar file

Commit:
Merge branch 'branch-New-Storage'

Commit:
Fix check style errors

Commit:
Adjust dialog box size

Commit:
Add window title

Commit:
fix width error

Commit:
Fix UI error

Commit:
Add data

Commit:
fix dialog box limit size

Commit:
data updating

Commit:
Merge branch 'branch-fix-ui-error'

# Conflicts:
#	data/List.txt
#	src/main/resources/view/DialogBox.fxml

Commit:
fix ui

Commit:
modify saved data

Commit:
Add a representative screenshot

Commit:
Merge branch 'branch-fix-date-format-bug'

# Conflicts:
#	data/List.txt
#	src/main/resources/view/DialogBox.fxml

Commit:
Add feature table

Commit:
Finishing user guide

Commit:
Merge branch 'branch-user-guide'

Commit:
Fix wrong read me File

Commit:
Reverse the original read me file

Commit:
Fix emoji error

Commit:
fix word

Commit:
Refactor

Commit:
Fix all remaining bugs and more testing

Commit:
Finalize

Commit:
Fix coding quality due to chat bot suggestion

april-anh committed:
Create _config.yml

Add theme to User Guide page

april-anh committed:
Create _config.yml

Add theme to User Guide page

april-anh committed:
Update _config.yml

april-anh committed:
Update README.md

april-anh committed:
Update README.md

april-anh committed:
Update README.md
